File: sdks/python/examples/create_inbox.py
"""Example: Creating an inbox with OpenAgentMail."""

import os

from openagentmail import OpenAgentMail

# Initialize client with API key from environment
client = OpenAgentMail(api_key=os.environ["OAM_API_KEY"])

# Create a new inbox
inbox = client.inboxes.create(
    username="support-agent",
    display_name="Support Agent",
    client_id="my-unique-inbox-id",  # For idempotency
)

print(f"Created inbox: {inbox.email}")
print(f"Inbox ID: {inbox.inbox_id}")
print(f"Created at: {inbox.created_at}")

# List all inboxes
print("\nAll inboxes:")
for inbox in client.inboxes.list():
    print(f"  - {inbox.email} ({inbox.inbox_id})")

File: sdks/typescript/examples/create-inbox.ts
// Example: Creating an inbox

import { OpenAgentMail } from 'openagentmail';

async function main() {
  const client = new OpenAgentMail({
    apiKey: process.env.OAM_API_KEY!,
  });

  // Create a new inbox
  const inbox = await client.inboxes.create({
    username: 'support',
    domain: 'example.com',
    displayName: 'Support Team',
    // Use clientId for idempotency
    clientId: 'support-inbox-v1',
  });

  console.log('Created inbox:', inbox.email);
  console.log('Inbox ID:', inbox.inboxId);

  // List all inboxes
  console.log('\nAll inboxes:');
  for await (const i of client.inboxes.list()) {
    console.log(`  - ${i.email} (${i.inboxId})`);
  }
}

main().catch(console.error);
